ext/gd/gd.c

```diff
@@ -4216,29 +4222,29 @@ static void _php_image_output_ctx(INTERNAL_FUNCTION_PARAMETERS, int image_type,
 
 	switch(image_type) {
 		case PHP_GDIMG_TYPE_JPG:
-			(*func_p)(im, ctx, q);
+			(*func_p)(im, ctx, quality);
 			break;
 		case PHP_GDIMG_TYPE_WEBP:
-			if (q == -1) {
-				q = 80;
+			if (quality == -1) {
+				quality = 80;
 			}
-			(*func_p)(im, ctx, q);
+			(*func_p)(im, ctx, quality);
 			break;
 		case PHP_GDIMG_TYPE_PNG:
-			(*func_p)(im, ctx, q, f);
+			(*func_p)(im, ctx, quality, basefilter);
 			break;
 		case PHP_GDIMG_TYPE_XBM:
 		case PHP_GDIMG_TYPE_WBM:
-			if (argc < 3) {
+			if (quality == -1) {
```

Comment:
This is confusing. For this functions, the parameter is `foreground`, so I think we should use a different variable. And in this case I'd also make it `l!`, as `-1` does not seem meaningful here.